sdk/core/core/inc/az_static_assert.h

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_STATIC_ASSERT_H
+#define AZ_STATIC_ASSERT_H
+
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AZ_STATIC_ASSERT(CONDITION) typedef int _az_static_assert[(CONDITION) ? 1 : -1];
+
+AZ_STATIC_ASSERT(true);
```

Comment:
can you static assert a true condition multiple times?

Reply:
yes, because the type definition will be the same.